### ISSUE TITLE
Non-stacked Extreme switches name ports differently internally vs LLDP

### ISSUE THREAD

**Issue by darpified** [librenms/librenms#4779]
Using Extremes with multiple versions of XOS, they represent through LLDP/EDP/CDP their ports as simple numeric entries i.e (1, 2, 49).
[links_ports.txt](https://github.com/librenms/librenms/files/524695/links_ports.txt)

But internally, they represent the ports as "1:1", as if they were in a stack of 1 switch.
[extreme_port.txt](https://github.com/librenms/librenms/files/524691/extreme_port.txt)

So, the auto-links don't work as they should, nor do the auto-maps.

I added a horrible kludge just to make sure I was right, but it's not adequate.
in discovery-protocols.inc.php in the LLDP section I did this:
`$broken_id = '1:'.$id;
                        $remote_port_id = dbFetchCell('SELECT`port_id`FROM`ports`WHERE (`ifName`= ? OR`ifDescr`= ? OR`ifName`= ? OR`ifDescr`= ? OR`ifName`= ? OR`ifPhysAddress`= ?) AND`device_id`= ?', array($broken_id, $if, $if, $id, $id, $remote_port_mac_address, $remote_device_id));
`
And that works as expected as far as the auto-links, but that's far from a "good" way to fix it.
I'd appreciate some guidance in the correct way to handle this, should it be considered a XOS bug and reported to Extreme?  Or should it just be coded around everywhere it occurs?

Thanks for any assistance,
RP.

DO NOT DELETE THIS INFORMATION.

> Please read this information carefully.
- [ ] Is your install up to date? [Updating your install](http://docs.librenms.org/General/Updating/)
    Please do not submit an issue if your install is not up to date within the last 24 hours or on a stable monthly release.
# \- [ ] Please include all of the information between the `====================================` section of `./validate.php`.

LibreNMS Version: 7ca80410d56d519d1353aae107bee715962ba46f
DB Schema: 144
PHP: 5.5.9-1ubuntu4.20
MySQL: 5.5.52-0ubuntu0.14.04.1
RRDTool: 1.4.7
# SNMP: NET-SNMP 5.7.2
- [ ] If you would like us to add a new device then please provide the information asked for [here](http://docs.librenms.org/Support/FAQ/#faq20)
- [ ] Please provide as much detail as possible.

**Comment by darpified** [comment]
Additionally, they have changed the "OS" name from "xos" to "extremeware"

**Comment by laf** [comment]
I think the best thing to do is provide the output of `snmpbulkwalk -OUneb -v2c -c COMMUNITY HOSTNAME .  | ./pbin.sh` for two devices that connect together over xDP for us to replicate this and work out the best way.

**Comment by laf** [comment]
No further info provided, please comment with links to the pbins and we can re-open.

**Comment by ellisway** [comment]
hi can this issue be reopened

i have a data set for 2 devices connected together running with lldp and edp located at the following address 

https://gist.github.com/ellisway/66570136d6af0da410d54d632ca5f79a

ellis

**Comment by murrant** [comment]
@ellisway Thanks for the walks I loaded them up.

Device A returns:
```
            [lldpRemChassisIdSubtype] => 4
            [lldpRemChassisId] => 00 04 96 28 06 AD
            [lldpRemPortIdSubtype] => 5
            [lldpRemPortId] => 26
            [lldpRemPortDesc] => ring-THE
            [lldpRemSysName] => thn-sw1
            [lldpRemSysDesc] => ExtremeXOS (X450e-24p) version 15.3.3.5 v1533b5-patch1-4 by release-manager on Sat Feb 15 02:26:00 EST 2014
            [lldpRemSysCapSupported] => 00 14
            [lldpRemSysCapEnabled] => 00 04
```

Device B returns:
```
            [ifIndex] => 1026
            [ifDescr] => X450e-24p Port 26
            [ifName] => 1:26
            [ifAlias] => ring-THE
            [ifType] => ethernetCsmacd
```


http://www.ieee802.org/1/files/public/MIBs/LLDP-MIB-200505060000Z.txt
```
            The enumeration 'interfaceName(5)' represents a port
            identifier based on the ifName MIB object, defined in IETF
            RFC 2863.
```

The switch says it return a lldpRemPortId type of 5 (which references ifName)

This doesn't seem to be an SNMP issue so much as an improper LLDP implementation.  The device should advertise the lldpRemPortId as the ifName `1:26`.  Imagine a stacked device.  I mean we could prepend 1: to all ifNames give by ExtremeXOS devices, but then the mapping would be wrong for stacked devices which is worse than no mapping in my opinion.

**Long story short, the manufacturer needs to fix this.**

The only thing LibreNMS could do is implement EDP support, but that only works if Extreme is the device on both ends.
